### ISSUE TITLE
Feature Request: Process bundles IDs that if focused block Nudge from opening UI

### ISSUE THREAD

**Issue by iJake** [macadmins/nudge#180]
If bundle IDs are provided AND they are running when Nudge attempts to open the UI, don't let it

Something  like:

JSON:
`{
	"userExperience": {
		"blockingAppBundleIDs": [
			"com.webex.meetingmanager",
			"us.zoom.xos"
		]
	}
}`

This should get bundle IDs of running processes:
`let runningApps = NSWorkspace.shared.runningApplications`

**Comment by erikng** [comment]
This feature is now merged in the development branch and I have tested with JSON. I really need someone to try with JSON and especially with mobileconfig.

**Comment by erikng** [comment]
I misinterpreted what you were wanting from this request.

Nudge already had logic for System Preferences / Major upgrades to stop nudging when those applications were top focused by the user. I assumed that you wanted to enhance this ability with your own list of applications. Others had requested this feature before, so I got confused.

What you are describing is if an application is simply running to not nudge. This will a bit of thinking on my part. While I already have a function that loops through the running applications, that is used for different purposes. Acceptable applications running are not hidden when getting into the "aggressive UI".

This would require extending that. I also think some companies (like mine) want the behavior I wrote, so the feature will require yet another key and additional logic. I am going to close this issue, but encourage you to write up another issue for that feature and if others want this, I will try and find time to write it. It will not make v1.1
